socialNetwork/socialNetwork/socialNetwork/Controllers/PostController.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using socialNetwork.Attributes;
+using socialNetwork.Models;
+using socialNetwork.Models.ViewModels;
+using socialNetwork.Paging;
+using socialNetwork.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace socialNetwork.Controllers
+{
+    //[ApiKey]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class PostController : ControllerBase
+    {
+        private readonly IPostService _postService;
+
+        public PostController(IPostService postService)
+        {
+            _postService = postService;
+        }
+
+        [HttpPost]
+        [Route("createPost/{groupId}/{userId}")]
+        public IActionResult CreatePost(Post post, int groupId, string userId)
+        {
+            var result = _postService.CreatePost(post, groupId, userId);
```

Comment:
Each user creates their own posts. One user cannot create a post for another. Therefore, you could try to change this and maybe some other routes and use the data from the token, i.e. from authorization, without sending userId.